test/e2e/checkpoint_test.go

```diff
@@ -725,17 +725,26 @@ var _ = Describe("Podman checkpoint", func() {
 		result.WaitWithDefaultTimeout()
 
 		Expect(result).Should(ExitCleanly())
-		Expect(result.OutputToString()).To(ContainSubstring(cid))
-		Expect(podmanTest.NumberOfContainersRunning()).To(Equal(0))
-		Expect(podmanTest.NumberOfContainers()).To(Equal(0))
+		Expect(result.OutputToString()).To(Equal(cid), "checkpoint output")
+		Expect(podmanTest.NumberOfContainersRunning()).To(Equal(0), "# of running containers after checkpoint")
+		// Allow a few seconds for --rm to take effect
+		ncontainers := podmanTest.NumberOfContainers()
+		for try := 0; try < 4; try++ {
+			if ncontainers == 0 {
+				break
+			}
+			time.Sleep(time.Second)
+			ncontainers = podmanTest.NumberOfContainers()
+		}
+		Expect(ncontainers).To(Equal(0), "# of containers (total) after checkpoint")
 
 		// Restore the container
 		result = podmanTest.Podman([]string{"container", "restore", "--ignore-rootfs", "-i", fileName})
 		result.WaitWithDefaultTimeout()
 
 		Expect(result).Should(ExitCleanly())
-		Expect(podmanTest.NumberOfContainersRunning()).To(Equal(1))
-		Expect(podmanTest.NumberOfContainers()).To(Equal(1))
+		Expect(podmanTest.NumberOfContainersRunning()).To(Equal(1), "# of running containers after restore")
+		Expect(podmanTest.NumberOfContainers()).To(Equal(1), "total # of containers after restore")
 		Expect(podmanTest.GetContainerStatus()).To(ContainSubstring("Up"))
```

Comment:
These checks are rather inefficient as each call to podmanTest... does a podman ps internally so all of these checks could really be combined into single podmn ps, to speed the test up. I know it is very little gain so not worth fixing now but at least for new tests I would call this a anti pattern.

Reply:
Did you mean to remove these as you added the one podman ps check above?

Reply:
Argh. Yes, of course I did. Thank you for catching that.